datalad/support/gitrepo.py

```diff
@@ -41,14 +116,14 @@ def __init__(self, path, url=None):
         Parameters
         ----------
         path: str
-          path to the git repository
+          path to the git repository; In case it's not an absolute path, it's relative to os.getcwd()
```

Comment:
In the case of constructors I don't see any alternative. What could possibly be another base for relative paths, as long as there is no repository object yet?
In general: let's see how it works for now as I don't see the problem nor a solution with no problems.